src/main/java/duke/dukeobject/Budget.java

```diff
@@ -0,0 +1,122 @@
+package duke.dukeobject;
+
+import duke.exception.DukeException;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.HashMap;
+import java.util.Scanner;
+
+public class Budget {
+
+
+    private static final String STORAGE_DELIMITER = "\n";
+
+    private File file;
+
+    private BigDecimal monthlyBudget;
+
+    /**
+     * Maps a category to the budget set for the category.
+     */
+    private HashMap<String, BigDecimal> budgetCategory;
+
+    /**
+     * Constructor of Budget object.
+     * @param file the File for budget in Duke
+     * @throws DukeException if the file cannot be created or read.
+     * @throws IOException if the file cannot be created or read.
+     */
+    public Budget(File file) throws DukeException, IOException {
+        this.file = file;
+        load();
+    }
+
+    /**
+     * Writes to the save file.
+     * @throws DukeException if unable to save the file successfully
+     */
+    public void save() throws DukeException {
+        try {
+            file.createNewFile();
+            try (FileWriter fileWriter = new FileWriter(file)) {
+                fileWriter.write(monthlyBudget.toPlainString());
+                if (!budgetCategory.isEmpty()) {
+                    for (String category : budgetCategory.keySet()) {
+                        BigDecimal budget = budgetCategory.get(category);
+                        fileWriter.write(category + " " + budget);
+                        fileWriter.write(STORAGE_DELIMITER);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new DukeException(String.format(DukeException.MESSAGE_SAVE_FILE_FAILED, file.getPath()));
+        }
+    }
+
+    /**
+     * loads from the save file.
+     * @throws DukeException if the file cannot be created or read.if the file cannot be created or read.
+     * @throws IOException if the file cannot be created or read.
+     */
+    public void load() throws DukeException, IOException {
+        file.createNewFile();
+        monthlyBudget = BigDecimal.ZERO;
+        budgetCategory = new HashMap<>();
+        try (Scanner fileReader = new Scanner(file).useDelimiter(STORAGE_DELIMITER)) {
+            if (fileReader.hasNext()) {
+                String monthlyBudgetString = fileReader.next();
+                monthlyBudget = new BigDecimal(monthlyBudgetString);
+            }
+            while (fileReader.hasNext()) {
+                String line = fileReader.next();
+                String[] separatedLine = line.split(" ");
+                String category = separatedLine[0];
+                String budgetString = separatedLine[1];
+                BigDecimal budget = new BigDecimal(budgetString);
+                budget.setScale(2, RoundingMode.HALF_UP);
```

Comment:
should actually write a parser for these and put into parser, as the parsing of BigDecimals corresponding to $ now varies between files. In my opinion, it should check if `bigDecimal.scale() <= 2` and throw an exception otherwise, rather than rounding decimals past the second.